ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLConversationalFlowAgentRunner.java

```diff
@@ -434,6 +434,12 @@ Map<String, String> getToolExecuteParams(MLToolSpec toolSpec, Map<String, String
             input = substitutor.replace(input);
             executeParams.put("input", input);
         }
+
+        // Override all parameters in tool config to tool execution parameters as the config contains the static parameters.
+        if (toolSpec.getConfigMap() != null && !toolSpec.getConfigMap().isEmpty()) {
+            executeParams.putAll(toolSpec.getConfigMap());
+        }
+
```

Comment:
The function is the same as MLFlowAgentRunner. Please extract the function to classes like `AgentUtils`.

Reply:
As far as I see, this PR writes all parameters from the new config map into the parameter map that is relevant for execution, which means that config parameters will overwrite existing ones with same name, i.e. have prioritization. Thus, specifying an own static input parameter as suggested in #2836 and #2918 is possible, but crucial is also the possibility to specify placeholders inside the new config. I can not see some substitutions here but maybe this takes place somewhere else, for all parameters?

For #2836, the placeholder `"${parameters.MLModelTool.output}"` was asked for.
For #2918, SearchIndexTool inside conversational agent, there is the need of a placeholder for the **last action input** of the agent, that must inserted into the OpenSearch query defined statically inside the new config. If conversational agent just sets the input parameter for the next tool, I doubt this information is lost if I specify a static input inside the tool config. 

Last but not least, the documentation should be adjusted accordingly, e.g. [SearchIndexTool](https://opensearch.org/docs/latest/ml-commons-plugin/agents-tools/tools/search-index-tool/) and [tutorial example](https://opensearch.org/docs/latest/ml-commons-plugin/tutorials/rag-conversational-agent/#exposing-only-the-question-parameter). Also some list or hint to the common available placeholders would be nice, in order to know e.g. how the placeholder for the former conversational agent step action input looks like.

Reply:
Good callout, I am thinking of supporting placeholders inside the config. Thanks @reuschling